### ISSUE TITLE
lib prefix on Windows

### ISSUE THREAD

**Issue by diorcety** [zeromq/zproject#942]
Is there any convention related to lib prefix for zproject on Windows.

Because by default in cmake for a library target called mlm on Windows the output file is mlm.dll and on linux it's libmlm.so

The issue is that project.libname is always prefixed by lib

project.libname is defined here
https://github.com/zeromq/zproject/blob/de9a6722f51e0784802a419bf574867ce8ae804a/zproject.gsl#L51

project.libname is used in python binding for finding the library

Shouldn't we use something like that?
```
set_target_properties(mlm PROPERTIES
    OUTPUT_NAME "mlm"
    PREFIX "lib")
```

**Comment by bluca** [comment]
I think one issue is backward compatibility - if the filename changes, there's a good chance all existing projects that use it will break

**Comment by jimklimov** [comment]
Well, the breakage mode needs that a project is regenerated. And users should review (git diff) or PR (and web-review) the changes. Also GSL can `echo` a warning about such change - though it does warn about many things that might bite you (they really can, so notice is not worthless).

Another breakage may be that projects which use your component (say mlm.dll as deliverable) can get broken by your upgrade and might have to rebuild. Or... as a legacy fallback you might add a rule (simple `cp` or `ln` equivalent, maybe optional) so you build and deliver both mlm.dll and libmlm.dll. For an ecosystem like our FTY project we can just rebuild all components and no longer deliver legacy filenames. Lower-level projects like mlm might have to serve both in `make install` and packaging on affected platforms.

_Sent from my Xiaomi Redmi Note 4 using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by bluca** [comment]
Yeah the problem is external reverse dependencies, that do not use zproject. Having a backward compatibility link could work as a solution

**Comment by diorcety** [comment]
About prefix I'm not sure but it seems that CMake don't put the prefix with MSVC not with MSYS or MINGW. It seems that nobody try the mlm/czmq python binding with MSVC toolchain.
